src/Microsoft.OpenApi/Writers/OpenApiWriterSettings.cs

```diff
@@ -24,5 +25,15 @@ internal bool ShouldInlineReference(OpenApiReference reference)
             return (reference.IsLocal && InlineLocalReferences)
                              || (reference.IsExternal && InlineExternalReferences);
         }
+
+        /// <summary>  
+        /// Enables sorting of collections using the default comparer  
+        /// </summary>  
+        public bool EnableSorting { get; set; }
+
+        /// <summary>  
+        /// Custom comparer for sorting collections.  
+        /// </summary>  
+        public IComparer<string>? KeyComparer { get; set; }
```

Comment:
we probably don't need the enable sorting flag, just the presence of a non-null value for the comparer should be enough